build-tools/packages/build-cli/src/commands/updates/packageTypes.ts

```diff
@@ -0,0 +1,157 @@
+	ALPHATRIMMED = "alphaTrimmedFilePath",
+	BETATRIMMED = "betaTrimmedFilePath",
+}
+
+/**
+ * A typeguard to check if a string is a {@link UpdatePackageJsonEnum}.
+ */
+function isUpdatePackageJsonTypes(type: string): type is UpdatePackageJsonEnum {
+	return (
+		type === UpdatePackageJsonEnum.ALPHA ||
+		type === UpdatePackageJsonEnum.BETA ||
+		type === UpdatePackageJsonEnum.DEFAULT
+	);
+}
+
+export default class UpdatePackageTypesCommand extends PackageCommand<
+	typeof UpdatePackageTypesCommand
+> {
+	static readonly description = "Updates the types in package.json based on the release flag";
+
+	static flags = {
+		path: Flags.directory({
+			description: "Path to a directory containing a package.",
+			exists: true,
+			required: true,
+		}),
+		types: Flags.custom<UpdatePackageJsonEnum>({
+			description:
+				"The types flag is used to specify the type of release when updating a package.json file. It accepts a custom enumeration called UpdatePackageJsonEnum that defines different release types.",
+			parse: async (input) => {
+				if (isUpdatePackageJsonTypes(input)) {
+					return input;
+				}
+				throw new Error(`Invalid release type: ${input}`);
+			},
+		})(),
+		...PackageCommand.flags,
+	};
+
+	private readonly packageTypesUpdated: string[] = [];
+
+	protected async processPackage(pkg: Package): Promise<void> {
+		// Define paths to package.json and api-extractor.json
+		const packageJsonFilePath = path.join(pkg.directory, "package.json");
+		const apiExtractorConfigFilePath = path.join(pkg.directory, "api-extractor.json");
+
+		// Check if api-extractor.json exists in the package directory
+		const apiExtractorConfigExists = fs.existsSync(apiExtractorConfigFilePath);
+
+		try {
+			// Read the content of the package.json file
+			const packageJsonContent = fs.readFileSync(packageJsonFilePath, "utf-8");
+
+			// Parse the JSON content to an object
+			const packageJson = JSON.parse(packageJsonContent);
+
+			if (apiExtractorConfigExists) {
+				// Read the content of api-extractor.json
+				const apiExtractorConfigContent = fs.readFileSync(
+					apiExtractorConfigFilePath,
+					"utf-8",
+				);
+
+				// Check if DTSROLLUP config exist in api-extractor.json
+				const isDtsRollupEnabled = apiExtractorConfigContent.includes(
+					ApiExtractorConfig.DTSROLLUP,
+				);
+				// Check if ALPHATRIMMED config exist in api-extractor.json
+				const isAlphaTrimmed = apiExtractorConfigContent.includes(
+					ApiExtractorConfig.ALPHATRIMMED,
+				);
+				// Check if BETATRIMMED config exist in api-extractor.json
+				const isBetaTrimmed = apiExtractorConfigContent.includes(
+					ApiExtractorConfig.BETATRIMMED,
+				);
+
+				if (isDtsRollupEnabled) {
+					const packageJsonUpdatePath = path.join(pkg.directory, "package.json");
+
+					// Extract the package name from the package directory
+					const directorySegments: string[] = pkg.directory.split("/");
+					const unscopedPackageName = directorySegments[directorySegments.length - 1];
```

Comment:
if you want the unscoped package name, this computation might not always be right. You can use pkg.nameUnscoped instead for a more robust (and already computed) approach derived from the package.json's name value.